Unskip torchaudio-related tests that are incompatible with torch 2.2.0

https://github.com/ludwig-ai/ludwig/actions/runs/6244093420/job/16950505282?pr=3644

@justinxzhao   I'm able to recreate the error with `test_audio_feature.py` on my end.
```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================================================================== short test summary info ============================================================================
FAILED ../../tests/ludwig/features/test_audio_feature.py::test_add_feature_data[raw] - RuntimeError
FAILED ../../tests/ludwig/features/test_audio_feature.py::test_add_feature_data[stft] - RuntimeError
FAILED ../../tests/ludwig/features/test_audio_feature.py::test_add_feature_data[stft_phase] - RuntimeError
FAILED ../../tests/ludwig/features/test_audio_feature.py::test_add_feature_data[group_delay] - RuntimeError
FAILED ../../tests/ludwig/features/test_audio_feature.py::test_add_feature_data[fbank] - RuntimeError
=================================================================== 5 failed, 6 passed, 7 warnings in 1.08s ====================================================================
```

Let me take a look and see what I can find out.

--- a/.github/workflows/pytest.yml
+++ b/.github/workflows/pytest.yml
@@ -78,7 +78,7 @@ jobs:
       - name: Setup Linux
         if: runner.os == 'linux'
         run: |
-          sudo apt-get update && sudo apt-get install -y cmake libsndfile1 wget
+          sudo apt-get update && sudo apt-get install -y cmake libsndfile1 wget libsox-dev
 
       - name: Setup macOS
         if: runner.os == 'macOS'

--- a/tests/ludwig/data/test_dataset_synthesizer.py
+++ b/tests/ludwig/data/test_dataset_synthesizer.py
@@ -1,14 +1,6 @@
-import pytest
-import torch
-from packaging import version
-
 from ludwig.data import dataset_synthesizer
 
 
-@pytest.mark.skipif(
-    version.parse(torch.__version__).base_version >= version.parse("2.2.0").base_version,
-    reason="Fails with torch 2.2.0. https://github.com/ludwig-ai/ludwig/issues/3645",
-)
 def test_build_synthetic_dataset(tmpdir):
     features = [
         {"name": "text", "type": "text"},

--- a/tests/ludwig/features/test_audio_feature.py
+++ b/tests/ludwig/features/test_audio_feature.py
@@ -5,7 +5,6 @@ from string import ascii_lowercase, ascii_uppercase, digits
 import pandas as pd
 import pytest
 import torch
-from packaging import version
 
 from ludwig.backend import LOCAL_BACKEND
 from ludwig.constants import BFILL, ENCODER_OUTPUT, PROC_COLUMN
@@ -53,10 +52,6 @@ def test_audio_input_feature(encoder: str) -> None:
     assert encoder_output[ENCODER_OUTPUT].shape[1:] == audio_input_feature.output_shape
 
 
-# @pytest.mark.skipif(
-#     version.parse(torch.__version__).base_version >= version.parse("2.2.0").base_version,
-#     reason="Fails with torch 2.2.0. https://github.com/ludwig-ai/ludwig/issues/3645",
-# )
 @pytest.mark.parametrize("feature_type", ["raw", "stft", "stft_phase", "group_delay", "fbank"])
 def test_add_feature_data(feature_type, tmpdir):
     preprocessing_params = {

--- a/tests/ludwig/features/test_audio_feature.py
+++ b/tests/ludwig/features/test_audio_feature.py
@@ -53,10 +53,10 @@ def test_audio_input_feature(encoder: str) -> None:
     assert encoder_output[ENCODER_OUTPUT].shape[1:] == audio_input_feature.output_shape
 
 
-@pytest.mark.skipif(
-    version.parse(torch.__version__).base_version >= version.parse("2.2.0").base_version,
-    reason="Fails with torch 2.2.0. https://github.com/ludwig-ai/ludwig/issues/3645",
-)
+# @pytest.mark.skipif(
+#     version.parse(torch.__version__).base_version >= version.parse("2.2.0").base_version,
+#     reason="Fails with torch 2.2.0. https://github.com/ludwig-ai/ludwig/issues/3645",
+# )
 @pytest.mark.parametrize("feature_type", ["raw", "stft", "stft_phase", "group_delay", "fbank"])
 def test_add_feature_data(feature_type, tmpdir):
     preprocessing_params = {

--- a/tests/ludwig/models/test_training_determinism.py
+++ b/tests/ludwig/models/test_training_determinism.py
@@ -3,8 +3,6 @@ import os
 
 import numpy as np
 import pytest
-import torch
-from packaging import version
 
 from ludwig.api import LudwigModel
 from ludwig.constants import BATCH_SIZE, EVAL_BATCH_SIZE, TRAINER
@@ -44,10 +42,6 @@ def test_training_determinism_ray_backend(csv_filename, tmpdir, ray_cluster_4cpu
     np.testing.assert_equal(train_stats_1, train_stats_2)
 
 
-@pytest.mark.skipif(
-    version.parse(torch.__version__).base_version >= version.parse("2.2.0").base_version,
-    reason="Fails with torch 2.2.0. https://github.com/ludwig-ai/ludwig/issues/3645",
-)
 def test_training_determinism_local_backend(csv_filename, tmpdir):
     experiment_output_1, experiment_output_2 = train_twice("local", csv_filename, tmpdir)
 

--- a/tests/regression_tests/model/test_old_models.py
+++ b/tests/regression_tests/model/test_old_models.py
@@ -43,16 +43,12 @@ def test_model_loaded_from_old_config_prediction_works(tmpdir):
     [
         "https://predibase-public-us-west-2.s3.us-west-2.amazonaws.com/ludwig_unit_tests/titanic_v07.zip",
         "https://predibase-public-us-west-2.s3.us-west-2.amazonaws.com/ludwig_unit_tests/twitter_bots_v05_1.zip",
-        # TODO(Justin): Audio features are broken for torch 2.2.
-        # "https://predibase-public-us-west-2.s3.us-west-2.amazonaws.com/ludwig_unit_tests/respiratory_v05.zip",
+        "https://predibase-public-us-west-2.s3.us-west-2.amazonaws.com/ludwig_unit_tests/respiratory_v05.zip",
         # TODO(Arnav): Re-enable once https://github.com/ludwig-ai/ludwig/issues/3150 is resolved since the GBM
         # model uses the PassthroughDecoder for the category output feature.
         # "https://predibase-public-us-west-2.s3.us-west-2.amazonaws.com/ludwig_unit_tests/gbm_adult_census_income_v061.zip",  # noqa: E501
     ],
-    ids=[
-        "titanic",
-        "twitter_bots",
-    ],  # "respiratory"],  # , "gbm_adult_census_income"],
+    ids=["titanic", "twitter_bots", "respiratory"],  # , "gbm_adult_census_income"],
 )
 def test_predict_deprecated_model(model_url, tmpdir):
     model_dir = os.path.join(tmpdir, "model")

